### PR TITLE
[Ready For Review] Unapologetic Vaurca Buffs II: Attack of the K'ois

### PR DESCRIPTION
**Pull request by LordFowl**
As per the request of @Bygonehero

Implements Black K'ois
Implements K'ois Mycosis
Implements Black K'ois Mycosis
Buffs K'ois' healing properties for Vaurca
Vaurca can now control where they bite

Also tidies up my last K'ois PR somewhat, primarily making it so that create_spores() does not proc if the tray's lid is shut, so botanists/xenobotanists can actually grow K'ois safely.

Black K'ois is a mutant variety of K'ois, meaning that currently it is only attainable via mutation of K'ois or adminbus. It functions almost identically to K'ois, except it exudes darkness instead of light, is black, and of course contains black k'ois spores instead of regular k'ois spores. It is also more tolerant of light changes since it naturally grows without light.

K'ois Mycosis has a chance of proc'ing instead of regular tox-loss whenever a non-vaurca consumes K'ois and if they do not already have a K'ois parasite. It creates a new subclass of organ, a parasite, inside of the target's chest, which grows over time. There are currently 4 stages with different effects, each separated by a 5 minute interval. (Each stage causes its effects and all the effects of the stages every Life()):

- Stage 1: Target is wracked with coughing, sometimes bloody, and occasionally feels stinging abdominal pains, which cause halloss.

- Stage 2: Target begins occasionally to gasp for air, and loses breath appropriately.

- Stage 3: The K'ois fully gestates, and the target begins to glow piss-yellow. Occasionally the K'ois inside of them will bear fruit, and they will be filled with phoron and k'ois paste as if they had eaten a K'ois.

- Stage 4: The K'ois is now fully mature, and the target will occasionally spew out clouds of K'ois spores, accompanied by vomiting, blood, and screaming.

Black K'ois Mycosis has a chance of proc'ing instead of regular tox-loss whenever a non-vaurca consumes Black K'ois and if they do not already have a Black K'ois parasite. It creates a new subclass of organ, a parasite, inside of the target's head, which grows over time. There are currently 4 stages with different effects, each separated by a 5 minute interval. (Each stage causes its effects and all the effects of the stages every Life()):

- Stages 1 & 2 are virtually identical to K'ois Mycosis as the Black K'ois spreads from the chest to the head. The target coughs and feels abdominal pain. Once stage 3 begins and the Black K'ois is fully settled in the cranium, these effects are replaced with head pain and disorientation, respectively.

- Stage 3: The K'ois fully gestates, and the target begins to reduce light like Black K'ois. Occasionally the K'ois inside of them will bear fruit, and they will be filled with trace amounts of phoron, but not as much as if they had eaten Black K'ois. At this point the subject will gain access to the Hivenet.

- Stage 4: The K'ois is now fully mature, and the target will occasionally spew out clouds of Black K'ois spores, accompanied by vomiting, blood, and screaming. They will also occasionally suffer brainloss until they die or the K'ois is removed. Most significantly however, the subject will become lobotomized, all of their memories will be wiped, they will only be able to speak hivenet, and all of their ambitions will be replaced with a desire to serve the Lii'dra hive. Due to the irrevocable nature of these changes, the only cure is death - removing the k'ois spore will only make them gibbering lobotomies.

Both of these infections can be cured by removing the parasite with regular organ removal surgery.

K'ois paste when consumed now heals 1.2 burn, brute, and tox per reagent when consumed by a Vaurca.

- [x] Testing
- [x] Balance
- [x] Changelog
- [x] Sprites for the parasites. Currently they just use the chestburster sprite.
~~- [ ] Sprites for K'ois infection overlays, and relevant code.~~ Saving for a later PR which makes such a system generally accessible without snowflake code.
- [x] Lore-dev approval @Bygonehero

### CODE REVIEW

File: code/modules/organs/subtypes/parasite.dm [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		if(prob(5*stage))
+			owner << "<span class='warning'>You feel a stinging pain in your abdomen!</span>"
+			owner.emote("me",1,"winces slightly.")
+			owner.adjustHalLoss(5)
+
+		if(prob(10))
+			owner.emote("cough")
+
+		if(prob(5))
+			spawn owner.emote("me", 1, "coughs up blood!")
```
Comment (Lohikar) [inline code/modules/organs/subtypes/parasite.dm]: I don't see why this needs to be spawned.

File: code/modules/organs/subtypes/parasite.dm [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+		else if(prob(5*stage))
+			owner << "<span class='warning'>You feel disorientated!</span>"
+			switch(rand(1,3))
+				if(1)
+					owner.confused += 10
+					owner.apply_effect(10,EYE_BLUR)
+				if(2)
+					owner.slurring += 30
+				if(3)
+					owner.dizziness += 10
```
Comment (Lohikar) [inline code/modules/organs/subtypes/parasite.dm]: Isn't this supposed to be done via. proc?

File: code/modules/organs/subtypes/parasite.dm [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+		if(stage < 3)
+			owner << "<span class='warning'>You feel a stinging pain in your abdomen!</span>"
+		else
+			owner << "<span class='warning'>You feel a stinging pain in your head!</span>"
+		owner.emote("me",1,"winces slightly.")
+		owner.adjustHalLoss(5)
+
+	if(stage >= 2)
+		if(stage < 3 && prob(10))
+			spawn owner.emote("me", 1, "gasps for air!")
```
Comment (Lohikar) [inline code/modules/organs/subtypes/parasite.dm]: Prob. doesn't need to be spawned.

File: code/modules/organs/subtypes/vaurca.dm [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+/obj/item/organ/vaurca/preserve/Initialize()
+	. = ..()
+
+	air_contents = new /datum/gas_mixture()
+	air_contents.adjust_gas("phoron", (ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C))
+	air_contents.volume = volume //liters
+	air_contents.temperature = T20C
+	distribute_pressure = ((pick(2.4,2.8,3.2,3.6)*ONE_ATMOSPHERE)*O2STANDARD)
+
+	START_PROCESSING(SSprocessing, src)
```
Comment (Lohikar) [inline code/modules/organs/subtypes/vaurca.dm]: Organs process already, this is probably doubling the process rate.

File: code/modules/organs/subtypes/vaurca.dm [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+	location.internal = src
+	usr << "<span class='notice'>You open \the [src] valve.</span>"
+	if (location.internals)
+		location.internals.icon_state = "internal1"
+
+	return
+
+/obj/item/organ/vaurca/preserve/Destroy()
+	if(air_contents)
+		qdel(air_contents)
```
Comment (Lohikar) [inline code/modules/organs/subtypes/vaurca.dm]: `QDEL_NULL`

File: code/modules/organs/subtypes/parasite.dm (hunk 1 above)
Comment (LordFowl) [reply code/modules/organs/subtypes/parasite.dm]: Should investigate lungs in general then, as I just copy-pasted with the assumption that there's some arcane technical reason you would spawn an emote()

File: code/modules/organs/subtypes/parasite.dm (hunk 2 above)
Comment (LordFowl) [reply code/modules/organs/subtypes/parasite.dm]: I don't think so?

**Comment by Bygonehero** [comment]
Looks great Fowl. I will see about sprites for the kois parasite.

File: code/modules/organs/subtypes/parasite.dm [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+			if(!(owner.species.flags & NO_PAIN))
+				owner.emote("scream")
+				owner.adjustHalLoss(15)
+				owner.drip(15)
+				owner.delayed_vomit()
+			return
+
+/obj/item/organ/parasite/blackkois/removed(var/mob/living/carbon/human/target)
+	if(all_languages[LANGUAGE_VAURCA] in target.languages && stage >= 3)
+		target.remove_language(LANGUAGE_VAURCA)
```
Comment (Arrow768) [inline code/modules/organs/subtypes/parasite.dm]: I think that would leave them with no language at all, as all other languages are removed during stage 4.

Comment (LordFowl) [reply code/modules/organs/subtypes/parasite.dm]: Correct. If the infection grows to stage 4 the only cure is cloning or cyborgification. By this point they've already been lobotomized by the K'ois.

Comment (Arrow768) [reply code/modules/organs/subtypes/parasite.dm]: If that is intentional, then its fine.

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 7]
```diff
@@ -255,21 +255,28 @@
...
 
-	last_special = world.time + 25
-
-	visible_message("<span class='warning'><b>\The [src]</b> rips viciously at \the [G.affecting]'s flesh with its mandibles!</span>")
-
 	if(istype(G.affecting,/mob/living/carbon/human))
 		var/mob/living/carbon/human/H = G.affecting
-		H.apply_damage(25,BRUTE, sharp=1, edge=1)
-		msg_admin_attack("[key_name_admin(src)] mandible'd [key_name_admin(H)] (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)",ckey=key_name(src),ckey_target=key_name(H))
+		var/mob/living/carbon/human/H2 = src
```
Comment (skull132) [inline code/modules/mob/living/carbon/human/human_powers.dm]: uh, wot? `src` is already of type `/mob/living/carbon/human`, why do you need to restate it as being that?

File: code/modules/organs/subtypes/parasite.dm [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+/obj/item/organ/parasite/kois
+	name = "k'ois mycosis"
+	icon_state = "kois-on"
+	dead_icon = "kois-off"
+
+	organ_tag = "kois"
+
+	parent_organ = "chest"
+
+/obj/item/organ/parasite/kois/process()
```
Comment (skull132) [inline code/modules/organs/subtypes/parasite.dm]: The structure of this proc looks weird. You shouldn't check `stage` for the same value _twice_. As in, you should only have one `if (stage >= 3)` on the same level. _Also_ consider if vs else-if.

File: code/modules/organs/subtypes/parasite.dm [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+	name = "k'ois mycosis"
+	icon_state = "black-on"
+	dead_icon = "black-off"
+	subtle = 1
+
+	organ_tag = "blackkois"
+
+	parent_organ = "head"
+
+/obj/item/organ/parasite/blackkois/process()
```
Comment (skull132) [inline code/modules/organs/subtypes/parasite.dm]: Same criticism here. The if conditions don't make sense.

File: code/modules/organs/subtypes/parasite.dm [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+
+		var/obj/item/organ/brain/B = owner.internal_organs_by_name["brain"]
+
+		if(B && !B.lobotomized)
+			owner << "<span class='danger'>As the K'ois consumes your mind, you feel your past self, your memories, your very being slip away... only slavery to the swarm remains...</span>"
+			owner << "<b>You have been lobotomized by K'ois infection. All of your previous memories up until this point are gone, and all of your ambitions are nothing. You live for only one purpose; to serve the Lii'dra hive.</b>"
+
+			B.lobotomized = 1
+
+		for(var/datum/language/L in owner.languages)
```
Comment (skull132) [inline code/modules/organs/subtypes/parasite.dm]: This is done every time the thing processes. This shouldn't be the case.

File: code/modules/organs/subtypes/vaurca.dm [hunk 11]
```diff
@@ -0,0 +1,317 @@
...
+	air_contents = new /datum/gas_mixture()
+	air_contents.adjust_gas("phoron", (ONE_ATMOSPHERE)*volume/(R_IDEAL_GAS_EQUATION*T20C))
+	air_contents.volume = volume //liters
+	air_contents.temperature = T20C
+	distribute_pressure = ((pick(2.4,2.8,3.2,3.6)*ONE_ATMOSPHERE)*O2STANDARD)
+
+	var/mob/living/carbon/location = loc
+
+	location.internal = src
+	usr << "<span class='notice'>You open \the [src] valve.</span>"
```
Comment (skull132) [inline code/modules/organs/subtypes/vaurca.dm]: \> `usr` in `Initialize`

wtf. no.

File: code/modules/organs/subtypes/vaurca.dm [hunk 12]
```diff
@@ -0,0 +1,317 @@
...
+			else
+				descriptive = "cold"
+		user << "<span class='notice'>\The [src] feels [descriptive].</span>"
+
+/obj/item/organ/vaurca/preserve/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	..()
+	var/obj/icon = src
+
+	if ((istype(W, /obj/item/device/analyzer)) && get_dist(user, src) <= 1)
+		for (var/mob/O in viewers(user, null))
```
Comment (skull132) [inline code/modules/organs/subtypes/vaurca.dm]: Why not use `user.visible_message()`?

File: code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm [hunk 13]
```diff
@@ -8,14 +8,66 @@
...
+	id = "blackkois"
+	description = "A thick goopy substance, rich in K'ois nutrients. This sample appears to be modified."
+	metabolism = REM * 4
+	var/nutriment_factor = 10
+	var/injectable = 0
+	color = "#31004A"
+	taste_description = "tar"
+	unaffected_species = IS_MACHINE
+
+/datum/reagent/blackkois/affect_ingest(var/mob/living/carbon/M, var/alien, var/removed)
```
Comment (skull132) [inline code/modules/reagents/Chemistry-Reagents/Chemistry-Reagents-Food-Drinks.dm]: `var/mob/living/carbon/human/H`

And just change line 50 to `if (istype(H))`

**Pull request merged**